code/modules/mob/living/simple_animal/bot/floorbot.dm

```diff
@@ -150,42 +157,42 @@
 	if(!..())
 		return
 
-	if(mode == BOT_REPAIRING)
+	if(mode == (BOT_REPAIRING || BOT_MAKE_TILE || BOT_EAT_TILE))
 		return
 
-	if(amount <= 0 && !target) //Out of tiles! We must refill!
-		if(eattiles) //Configured to find and consume floortiles!
-			target = scan(/obj/item/stack/tile/plasteel)
-			process_type = null
-
-		if(!target && maketiles) //We did not manage to find any floor tiles! Scan for metal stacks and make our own!
-			target = scan(/obj/item/stack/sheet/metal)
-			process_type = null
-			return
-		else
-			if(nag_on_empty) //Floorbot is empty and cannot acquire more tiles, nag the engineers for more!
-				nag()
-
 	if(prob(5))
 		audible_message("[src] makes an excited booping beeping sound!")
 
 	//Normal scanning procedure. We have tiles loaded, are not emagged.
-	if(!target && emagged < 2 && amount > 0)
+	if(!target && emagged < 2 && amount)
 		if(!target)
 			process_type = HULL_BREACH //Ensures the floorbot does not try to "fix" space areas or shuttle docking zones.
-			target = scan(/turf/space)
+			target = scan(/turf/space, avoid_bot = /mob/living/simple_animal/bot/floorbot)
 
-		if(!target && replacetiles) //Finds a floor without a tile and gives it one.
+		if(!target && replace_tiles) //Finds a floor without a tile and gives it one.
 			process_type = REPLACE_TILE //The target must be the floor and not a tile. The floor must not already have a floortile.
-			target = scan(/turf/simulated/floor)
+			target = scan(/turf/simulated/floor, avoid_bot = /mob/living/simple_animal/bot/floorbot)
 
-		if(!target && fixfloors) //Repairs damaged floors and tiles.
+		if(!target && fix_floor) //Repairs damaged floors and tiles.
 			process_type = FIX_TILE
-			target = scan(/turf/simulated/floor)
+			target = scan(/turf/simulated/floor, avoid_bot = /mob/living/simple_animal/bot/floorbot)
 
 	if(!target && emagged == 2) //We are emagged! Time to rip up the floors!
 		process_type = TILE_EMAG
-		target = scan(/turf/simulated/floor)
+		target = scan(/turf/simulated/floor, avoid_bot = /mob/living/simple_animal/bot/floorbot)
+
+	if(amount < MAX_AMOUNT && !target) //Out of tiles! We must refill!
+		if(eat_tiles) //Configured to find and consume floortiles!
+			target = scan(/obj/item/stack/tile/plasteel)
+			process_type = null
+
+		if(!target && make_tiles) //We did not manage to find any floor tiles! Scan for metal stacks and make our own!
+			target = scan(/obj/item/stack/sheet/metal)
+			process_type = null
+			return
+		else
+			if(nag_on_empty) //Floorbot is empty and cannot acquire more tiles, nag the engineers for more!
```

Comment:
```suggestion
			if(!target && nag_on_empty) //Floorbot is empty and cannot acquire more tiles, nag the engineers for more!
```

Reply:
This should be fine?
As comes if `if(!target && make_tiles)` fails, should be fine?

Reply:
Oops, I meant with this one that you should get rid of the else and add a !target to nag on empty

Reply:
```
		if(!target && make_tiles) //We did not manage to find any floor tiles! Scan for metal stacks and make our own!
			target = scan(/obj/item/stack/sheet/metal)
			process_type = null
		else
			if(nag_on_empty) //Floorbot is empty and cannot acquire more tiles, nag the engineers for more!
				nag()
```
to
```
		if(!target && make_tiles) //We did not manage to find any floor tiles! Scan for metal stacks and make our own!
			target = scan(/obj/item/stack/sheet/metal)
			process_type = null
			
		if(!target && nag_on_empty) //Floorbot is empty and cannot acquire more tiles, nag the engineers for more!
			nag()
```
?

Reply:
yeah